domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/ValidPartAdmissionConstraintValidator.java

```diff
@@ -39,6 +45,56 @@ public boolean isValid(PartAdmissionResponse response, ConstraintValidatorContex
             valid = false;
         }
 
+        if (isDefendantIndividual(response.getDefendant())
+            && !isPaymentOptionImmediately(response)
+            && !isStatementOfMeansPopulated(response)) {
+            setValidationErrorForMandatoryStatementOfMeans(context);
+            valid = false;
+        }
+
+        if (isStatementOfMeansPopulated(response)
+            && (isPaymentOptionImmediately(response) || !isDefendantIndividual(response.getDefendant()))) {
+            setValidationErrorForPopulatedStatementOfMeans(context);
+            valid = false;
+        }
+
         return valid;
     }
+
+    private static boolean isStatementOfMeansPopulated(PartAdmissionResponse response) {
+        return response.getStatementOfMeans().isPresent();
+    }
+
+    private static boolean isDefendantIndividual(Party defendant) {
+        return defendant instanceof Individual || defendant instanceof SoleTrader;
+
+    }
+
+    private static boolean isPaymentOptionImmediately(PartAdmissionResponse response) {
+        return response.getPaymentIntention().isPresent()
+            && !response.getPaymentIntention().get().getPaymentOption().equals(PaymentOption.IMMEDIATELY);
+    }
+
+    private static void setValidationErrorForMandatoryStatementOfMeans(ConstraintValidatorContext validatorContext) {
```

Comment:
I would have inlined this as name does not make it more readable. Neither there is need to have it as methods for reusability point of view.